### ISSUE TITLE
Default comparer incorrectly reports trees as equal

### ISSUE THREAD

**Issue by RobSwDev** [ValeraT1982/ObjectsComparer#9]
The simple test case below fails. Tried debugging it and didn't see anything glaringly obvious. 
Looks like an incredibly useful project otherwise.
 
```
   [TestFixture]
    public class TreeTests
    {
        [Test]
        public void RobTreeDiff()
        {
            var t1 = T(1, T(2));
            var t2 = T(1, T(2, T(3)));

            AssertObjectsDiffer(t1, t2);
        }

        public static ITreeNode<TData> T<TData>(TData data, params ITreeNode<TData>[] children)
        {
            var node = new TreeNode<TData>(data);
            foreach (var c in children)
                node.ChildNodes.Add(c);
            return node;
        }

        public static void AssertObjectsDiffer<T>(T expected, T actual)
        {
            var comparer = new Comparer<T>();
            Assert.IsFalse(comparer.Compare(expected, actual, out var differences));
        }
    }

    public interface ITreeNode<TData>
    {
        TData Data { get; }

        IList<ITreeNode<TData>> ChildNodes { get; }
    }

    public class TreeNode<T> : ITreeNode<T>
    {
        public T Data { get; }

        public IList<ITreeNode<T>> ChildNodes { get; }

        public TreeNode(T node)
        {
            this.Data = node;
            this.ChildNodes = new List<ITreeNode<T>>();
        }
    }
```

**Comment by ValeraT1982** [comment]
@RobSwDev good catch. Comparer doesn't compere properties of the parent interfaces. It will be fixed in the next version.
For now the easiest way to fix it is to change ChildNodes type from IList<....> to List<>.

**Comment by ValeraT1982** [comment]
Fixed in 1.3.1
